AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.appwidget
+
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.util.SizeF
+import android.widget.RemoteViews
+
+/**
+ * Implementation of the weather forecast app widget that demonstrates the flexible layouts based
+ * on the size of the device.
+ */
+class WeatherForecastAppWidget : AppWidgetProvider() {
+
+    override fun onUpdate(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetIds: IntArray
+    ) {
+        for (appWidgetId in appWidgetIds) {
+            updateWeatherWidget(context, appWidgetManager, appWidgetId)
+        }
+    }
+
+    private fun updateWeatherWidget(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetId: Int
+    ) {
+        val viewMapping: MutableMap<SizeF, RemoteViews> = mutableMapOf()
+        // Specify the maximum width and height in dp and a layout, which you want to use for the
+        // specified size
```

Comment:
I couldn't understand what this comment means. Does the system choose one of these instances of RemoteViews depending on the available area?

Reply:
The system picks up the layout depending on the widget size being resized.
Added more explanation about that.

Reply:
The comment says "maxium width and height", but the explanation below seems to say that they are minimum size?

Reply:
Oh that's right. Fixed the comment, thanks.